### DISCUSSION TITLE
Bump Mac Casks on Linux

### DISCUSSION THREAD

**Discussion by WangEdward** [Homebrew/discussions#5288] [Linux]
## The problem:

Since homebrew `4.2.0`, calling `OS::Mac` on Linux is deprecated in https://github.com/Homebrew/brew/pull/16224 and https://github.com/Homebrew/brew/pull/16306

I used to use Linux machines on GitHub actions to **bump some Mac casks** in a **private** repo. This no longer functioned due to the error: `Calling 'MacOS.version' on Linux is deprecated! There is no replacement.` 

## Current workaround

I manually checked out homebrew to `tags/4.1.21` in my workflow file. However, I don't think it's a good idea to use an outdated version all the time.

**Why do I not use Mac machines provided by GitHub action?** Mac machines are billed 10x for private repos. Using Linux machines allows me to run the bump more frequently.

## My question:

Is it still possible to run `brew bump` for Mac casks on Linux?

If not, is there any alternative way to bump a cask without switching to Mac / stuck on an old version?

---

Thanks for the help! ❤️

**Comment by SMillerDev** [comment]
Have you tried `HOMEBREW_SIMULATE_MACOS_ON_LINUX`?

https://docs.brew.sh/Manpage#environment

**Reply by WangEdward** [reply]
Thanks for your help!

I tried to add `HOMEBREW_SIMULATE_MACOS_ON_LINUX`, but sadly it didn't seem to make any difference. 😭

<details><summary>workflow log:</summary>

```
Run Homebrew/actions/bump-packages@master
  with:
    token: ***
    casks: ***
  
    fork: false
  env:
    CASKS: ***
  
    HOMEBREW_NO_INSTALL_FROM_API: 
    HOMEBREW_SIMULATE_MACOS_ON_LINUX: 1
Run if [[ "$INPUT_FORK" == "true" ]]; then
  if [[ "$INPUT_FORK" == "true" ]]; then
    brew bump --open-pr --casks ***
  
  else
    brew bump --no-fork --open-pr --casks ***
  
  fi
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    CASKS: ***
  
    HOMEBREW_NO_INSTALL_FROM_API: 
    HOMEBREW_SIMULATE_MACOS_ON_LINUX: 1
    HOMEBREW_DEVELOPER: 1
    HOMEBREW_GITHUB_API_TOKEN: ***
==> Installing 'bundler' gem
Fetching gem metadata from https://rubygems.org/.......
Fetching bindata 2.5.0
Fetching public_suffix 5.0.5
Fetching rexml 3.2.6
Fetching plist 3.7.1
Installing bindata 2.5.0
Installing plist 3.7.1
Installing rexml 3.2.6
Installing public_suffix 5.0.5
Fetching ruby-macho 4.0.1
Installing ruby-macho 4.0.1
Fetching sorbet-runtime 0.5.11346
Fetching ruby-progressbar 1.13.0
Fetching warning 1.3.0
Installing ruby-progressbar 1.13.0
Installing warning 1.3.0
Fetching addressable 2.8.6
Fetching elftools 1.3.0
Installing sorbet-runtime 0.5.11346
Installing elftools 1.3.0
Installing addressable 2.8.6
Fetching patchelf 1.5.0
Installing patchelf 1.5.0
Bundle complete! 35 Gemfile dependencies, 12 gems now installed.
Bundled gems are installed into `../../../../linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle`
Error: Calling `MacOS.version` on Linux is deprecated! There is no replacement.
Error: Calling `MacOS.version` on Linux is deprecated! There is no replacement.
```
</details>

**Comment by ipatch** [comment]
i got bit by this today to, when setting up a ubuntu runner for the freecad tap. the below links helped shed some light on my issue. hope they help.

https://github.com/Homebrew/brew/issues/15343

https://docs.brew.sh/Formula-Cookbook#handling-different-system-configurations
